ExampleMod/Content/Projectiles/ExampleJoustingLanceProjectile.cs

```diff
@@ -0,0 +1,210 @@
+			Projectile.hide = true; // We are drawing the projectile ourself. See PreDraw() below.
+			Projectile.ownerHitCheck = true; // Make sure the owner of the projectile has line of sight to the target (aka can't hit things through tile).
+			Projectile.DamageType = DamageClass.MeleeNoSpeed; // Set the damage to melee damage.
+
+			// Act like the normal Jousting Lance. Use this if you set the aiStyle to 19.
+			// AIType = ProjectileID.JoustingLance; 
+		}
+
+		// This is the behavior of the Jousting Lances.
+		public override void AI() {
+			Player owner = Main.player[Projectile.owner]; // Get the owner of the projectile.
+			Vector2 center = owner.RotatedRelativePoint(owner.MountedCenter); // Get the center of the owner.
+			Projectile.direction = owner.direction; // Set the direction of the projectile to the same direction as the owner.
+			owner.heldProj = Projectile.whoAmI; // Set the owner's held projectile to this projectile.
+			owner.itemTime = owner.itemAnimation; // Set the time left on this item to the animation time.
+			Projectile.Center = center; // Set the center of the projectile to the center of the owner. Projectile.Center is now actually the tip of the Jousting Lance.
+
+			int itemAnimationMax = owner.itemAnimationMax;
+			int itemAnimation = owner.itemAnimation;
+
+			// Changing this denominator will affect how far the projectile can move when it is shot (based on the scaleFactor).
+			// It is recommend to keep it between 2 and 12. 13 and above will kill the projectile. 1 will make it not move forward at all. Vanilla uses 3.
+			int itemAnimationMaxDiv = itemAnimationMax / 3;
+
+			float smallerItemAnimation = MathHelper.Min(itemAnimation, itemAnimationMaxDiv); // Choose the smaller of the two values.
+			float minimumItemAnimation = itemAnimation - smallerItemAnimation;
+			float spawnDistance = 28f; // This changes how far out the projectile is spawned. Change this value if your lance is longer or shorter. Vanilla uses 28f.
+			float flyOut = 0.4f; // This changes how far out the projectile flies when it spawns. Vanilla uses 0.4f
+			float flyBack = 0.4f; // This changes how far back the projectile flies when it is killed. Vanilla uses 0.4f
+
+			// Fade the projectile in when it first spawns
+			Projectile.alpha -= 40;
+			if (Projectile.alpha < 0) {
+				Projectile.alpha = 0;
+			}
+
+			float pushOutMulti = itemAnimationMax - itemAnimationMaxDiv - minimumItemAnimation; // This value will get larger the further out the projectile is from the player until it reaches it's maximum distance.
+			float pullInMulti = itemAnimationMaxDiv - smallerItemAnimation; // This value will get larger when the projectile is being pulled back in.
+			float scaleFactor = spawnDistance + flyOut * pushOutMulti - flyBack * pullInMulti;
+			Projectile.position += Projectile.velocity * scaleFactor; // Move the projectile based on the velocity and scaleFactor
```

Comment:
The names and math here is quite confusing. Especially because `Item.shootSpeed` throws a `3.5f` multiplier in. Also you have a mixture of alpha code, channel time code and tip positioning code all together.

I suggest changing `Item.shootSpeed` to `1f` and using code like below:
```cs
			int itemAnimationMax = owner.itemAnimationMax;
			// Remember, frames count down from itemAnimationMax to 0
			// Frame at which the lance is fully extended. Hold at this frame before retracting.
			// Scale factor (0.34f) means the last 34% of the animation will be used for retracting.
			int holdOutFrame = (int)(itemAnimationMax * 0.34f);
			if (owner.channel && owner.itemAnimation < holdOutFrame) {
				owner.SetDummyItemTime(holdOutFrame); // This makes it so the projectile never dies while we are holding it (except when we take damage, see ExampleJoustingLancePlayer).
			}

			// If the Jousting Lance is no longer being used, kill the projectile.
			if (owner.ItemAnimationEndingOrEnded) {
				Projectile.Kill();
				return;
			}

			int itemAnimation = owner.itemAnimation;
			// extension and retraction factors (0-1). As the animation plays out, extension goes from 0-1 and stays at 1 while holding, then retraction goes from 0-1.
			float extension = 1 - Math.Max(itemAnimation - holdOutFrame, 0) / (float)(itemAnimationMax - holdOutFrame);
			float retraction = 1 - Math.Min(itemAnimation, holdOutFrame) / (float)holdOutFrame;

			// distances are in pixels
			float extendDist = 22; // How far to fly out during extension
			float retractDist = extendDist / 2; // How far to fly back during retraction
			float tipDist = 98 + extension * extendDist - retraction * retractDist;

			Vector2 center = owner.RotatedRelativePoint(owner.MountedCenter); // Get the center of the owner.
			Projectile.Center = center; // Set the center of the projectile to the center of the owner. Projectile.Center is now actually the tip of the Jousting Lance.
			Projectile.position += Projectile.velocity * tipDist; // The projectile velocity contains the orientation of the lance, multiply it by the tipDist to position the tip

			// Set the rotation of the projectile.
			// For reference, 0 is the top left, 180 degrees or pi radians is the bottom right.
			Projectile.rotation = (float)Math.Atan2(Projectile.velocity.Y, Projectile.velocity.X) + (float)Math.PI * 3 / 4f;

			// Fade the projectile in when it first spawns
			Projectile.alpha -= 40;
			if (Projectile.alpha < 0) {
				Projectile.alpha = 0;
			}
```

Reply:
Yeah you are right. I just copied and pasted the decompiled source code and explained what it was doing. It didn't even cross my mind to refactor the code to make it more understandable.